inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/special/ScheduledExecutorContextPropagationSensor.java

```diff
@@ -0,0 +1,82 @@
+package rocks.inspectit.oce.core.instrumentation.special;
+
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.asm.AsmVisitorWrapper;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.dynamic.DynamicType;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.context.annotation.DependsOn;
+import org.springframework.stereotype.Component;
+import rocks.inspectit.oce.bootstrap.Instances;
+import rocks.inspectit.oce.core.instrumentation.config.model.InstrumentationConfiguration;
+import rocks.inspectit.oce.core.instrumentation.context.ContextManagerImpl;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+/**
+ * Special sensor for passing the context via the {@link ScheduledExecutorService}.
+ * This sensor will pass the context when using the following methods:
+ * <p><ul>
+ * <li>{@link ScheduledExecutorService#schedule(Runnable, long, TimeUnit)}
+ * <li>{@link ScheduledExecutorService#schedule(Callable, long, TimeUnit)}
+ * <li>{@link ScheduledExecutorService#scheduleWithFixedDelay(Runnable, long, long, TimeUnit)}
+ * <li>{@link ScheduledExecutorService#scheduleAtFixedRate(Runnable, long, long, TimeUnit)}
+ * </ul><p>
+ */
+@Component
+@DependsOn(ContextManagerImpl.BEAN_NAME)
+public class ScheduledExecutorContextPropagationSensor implements SpecialSensor {
+
+    private static final ElementMatcher<TypeDescription> CLASSES_MATCHER = isSubTypeOf(ScheduledExecutorService.class);

```

Comment:
I am also getting this, seems wrong:

`2019-01-25 12:07:46,389 DEBUG 2623   --- [inspectIT] [pectit-thread-0] r.i.o.c.i.AsyncClassTransformer          : Applying instrumentation of interface java.util.concurrent.ScheduledExecutorService`

maybe it's the Java 8 and default interface methods?

Reply:
Should also be covered by excluding the abstract classes.